### PR TITLE
Update various calls to mirrorlist.centos.org with vault.centos.org

### LINKED ISSUE #3905: Update CentOS mirrorlink
https://ci.adoptium.net/job/Test_openjdk23_hs_special.system_x86-64_linux/7/

Current test is trying to use CentOS 7 which no longer exists

```
docker run -v "/home/jenkins/workspace/Test_openjdk23_hs_special.system_x86-64_linux_rerun/aqa-tests/TKG/../../jvmtest/system/reproducibleCompare:/home/jenkins/test" -w "/home/jenkins/" -v "/home/jenkins/workspace/Test_openjdk23_hs_special.system_x86-64_linux_rerun/jdkbinary/j2sdk-image/jdk-23+35/bin/../:/home/jenkins/jdkbinary/" --name reproducibleCompare centos:7 /bin/bash /home/jenkins/test/linux_repro_build_compare.sh /home/jenkins/test/OpenJDK23U-sbom_x64_linux_hotspot_23_35-ea.json /home/jenkins/jdkbinary; \
[2024-08-02T04:48:53.423Z] if [ $? -eq 0 ]; then echo "-----------------------------------"; echo "Rebuild_Same_JDK_Reproducibility_Test_0""_PASSED"; echo "-----------------------------------"; cd /home/jenkins/workspace/Test_openjdk23_hs_special.system_x86-64_linux_rerun/aqa-tests/TKG/..; rm -f -r "/home/jenkins/workspace/Test_openjdk23_hs_special.system_x86-64_linux_rerun/aqa-tests/TKG/../TKG/output_17225741323010/Rebuild_Same_JDK_Reproducibility_Test_0"; else echo "-----------------------------------"; echo "Rebuild_Same_JDK_Reproducibility_Test_0""_FAILED"; echo "-----------------------------------"; fi; \
[2024-08-02T04:48:53.423Z] docker cp reproducibleCompare:/home/jenkins/reprotest.diff ./; \
[2024-08-02T04:48:53.423Z] docker cp reproducibleCompare:/home/jenkins/reproJDK.tar.gz ./; \
[2024-08-02T04:48:53.423Z] docker cp reproducibleCompare:/home/jenkins/SBOM.json ./; \
[2024-08-02T04:48:53.423Z] docker container rm reproducibleCompare; \
[2024-08-02T04:48:53.423Z] echo "";	echo "TEST TEARDOWN:"; \
[2024-08-02T04:48:53.423Z] echo "Nothing to be done for teardown."; \
[2024-08-02T04:48:53.423Z]  } 2>&1 | tee -a "/home/jenkins/workspace/Test_openjdk23_hs_special.system_x86-64_linux_rerun/aqa-tests/TKG/../TKG/output_17225741323010/TestTargetResult";
[2024-08-02T04:48:53.423Z] 
[2024-08-02T04:48:53.423Z] TEST SETUP:
[2024-08-02T04:48:53.423Z] Nothing to be done for setup.
[2024-08-02T04:48:53.423Z] 
[2024-08-02T04:48:53.423Z] TESTING:
[2024-08-02T04:48:54.091Z] Error: No such container: reproducibleCompare
[2024-08-02T04:48:56.260Z] Loaded plugins: fastestmirror, ovl
[2024-08-02T04:48:56.260Z] Determining fastest mirrors
[2024-08-02T04:48:56.260Z] Could not retrieve mirrorlist http://mirrorlist.centos.org/?release=7&arch=x86_64&repo=os&infra=container error was
[2024-08-02T04:48:56.260Z] 14: curl#6 - "Could not resolve host: mirrorlist.centos.org; Unknown error"
[2024-08-02T04:48:56.260Z] 
[2024-08-02T04:48:56.260Z] 
[2024-08-02T04:48:56.260Z]  One of the configured repositories failed (Unknown),
[2024-08-02T04:48:56.260Z]  and yum doesn't have enough cached data to continue. At this point the only
[2024-08-02T04:48:56.260Z]  safe thing yum can do is fail. There are a few ways to work "fix" this:
[2024-08-02T04:48:56.260Z] 
[2024-08-02T04:48:56.260Z]      1. Contact the upstream for the repository and get them to fix the problem.
[2024-08-02T04:48:56.260Z] 
[2024-08-02T04:48:56.260Z]      2. Reconfigure the baseurl/etc. for the repository, to point to a working
[2024-08-02T04:48:56.260Z]         upstream. This is most often useful if you are using a newer
[2024-08-02T04:48:56.260Z]         distribution release than is supported by the repository (and the
[2024-08-02T04:48:56.260Z]         packages for the previous distribution release still work).
[2024-08-02T04:48:56.260Z] 
[2024-08-02T04:48:56.260Z]      3. Run the command with the repository temporarily disabled
[2024-08-02T04:48:56.260Z]             yum --disablerepo=<repoid> ...
[2024-08-02T04:48:56.260Z] 
[2024-08-02T04:48:56.260Z]      4. Disable the repository permanently, so yum won't use it by default. Yum
[2024-08-02T04:48:56.260Z]         will then just ignore the repository until you permanently enable it
[2024-08-02T04:48:56.260Z]         again or use --enablerepo for temporary usage:
[2024-08-02T04:48:56.260Z] 
[2024-08-02T04:48:56.260Z]             yum-config-manager --disable <repoid>
[2024-08-02T04:48:56.260Z]         or
[2024-08-02T04:48:56.260Z]             subscription-manager repos --disable=<repoid>
[2024-08-02T04:48:56.260Z] 
[2024-08-02T04:48:56.260Z]      5. Configure the failing repository to be skipped, if it is unavailable.
[2024-08-02T04:48:56.260Z]         Note that yum will try to contact the repo. when it runs most commands,
[2024-08-02T04:48:56.260Z]         so will have to try and fail each time (and thus. yum will be be much
[2024-08-02T04:48:56.260Z]         slower). If it is a very temporary problem though, this is often a nice
[2024-08-02T04:48:56.260Z]         compromise:
[2024-08-02T04:48:56.260Z] 
[2024-08-02T04:48:56.260Z]             yum-config-manager --save --setopt=<repoid>.skip_if_unavailable=true
[2024-08-02T04:48:56.260Z] 
[2024-08-02T04:48:56.260Z] Cannot find a valid baseurl for repo: base/7/x86_64
[2024-08-02T04:48:56.928Z] -----------------------------------
[2024-08-02T04:48:56.928Z] Rebuild_Same_JDK_Reproducibility_Test_0_FAILED
[2024-08-02T04:48:56.928Z] -----------------------------------
```

### PR DESCRIPTION
**Pull request by sophia-guo**
Mirrorlist has been deprecated following the EOL of Centos 7

Fix #3905

### CODE REVIEW

**Comment by sophia-guo** [comment]
Rerun job https://ci.adoptium.net/job/Test_openjdk23_hs_special.system_x86-64_linux_testList_0/11/ with this PR.

**Comment by sophia-guo** [comment]
Note there are two agents available. https://ci.adoptium.net/label/ci.role.test&&hw.arch.x86&&sw.os.linux&&!%28centos6%7C%7Crhel6%29&&sw.tool.docker/
However with https://ci.adoptium.net/computer/test-ibmcloud-rhel7-x64-1/ will consistently hit the issue with error message like

```
10:40:42  docker: failed to register layer: symlink ../a0b6a018cc497a3d95ea4f449e24cd883f3826647b7c3b469f05d683e56907ee/diff /var/lib/docker/overlay2/l/XVAC3AEAM6DZ4IXSKPGV2D7DZC: no such file or directory.
10:40:42  See 'docker run --help'.
```
https://ci.adoptium.net/job/Test_openjdk23_hs_special.system_x86-64_linux_testList_0/14/console

Which suggests the docker issue and need to restart the docker. It is not related with this PR, will open an separate issue in infra repo.

**Comment by sxa** [comment]
> https://ci.adoptium.net/job/Test_openjdk23_hs_special.system_x86-64_linux_testList_0/14/console
> 
> Which suggests the docker issue and need to restart the docker. It is not related with this PR, will open an separate issue in infra repo.

I've fixed that issue and am doing a "re-run in grinder" of your job at https://ci.adoptium.net/job/Grinder/10763/

**Review by sxa — APPROVED** [review]
> I've fixed that issue and am doing a "re-run in grinder" of your job at https://ci.adoptium.net/job/Grinder/10763/

Approving on the basis that it seems to be running through the build ok now so it's likely to complete ok :-)

File: tooling/reproducible/linux_repro_build_compare.sh [hunk 1]
```diff
@@ -28,6 +28,9 @@ isJdkDir=false
 USING_DEVKIT="false"
 installPrereqs() {
   if test -r /etc/redhat-release; then
+    #Replace mirrorlist to vault as centos7 reached EOL.
```
Comment (karianna) [inline tooling/reproducible/linux_repro_build_compare.sh:31]:
```suggestion
    # Replace mirrorlist to vault as centos7 reached EOL.
```

**Review by karianna — APPROVED** [review]

**Pull request merged**